cdap-data-fabric/src/main/java/io/cdap/cdap/spi/data/sql/RetryingSqlTransactionRunner.java

```diff
@@ -72,25 +82,40 @@ public void run(TxRunnable runnable) throws TransactionException {
         if (TRANSACTION_CONFLICT_SQL_STATE.equals(sqlState)) {
           metricsCollector.increment(Constants.Metrics.StructuredTable.TRANSACTION_CONFLICT, 1L);
           ++retries;
-          long delay = retries > MAX_RETRIES ? -1 : DELAY_MILLIS;
-          if (delay < 0) {
-            throw e;
-          }
-
-          if (delay > 0) {
-            try {
-              TimeUnit.MILLISECONDS.sleep(delay);
-            } catch (InterruptedException e1) {
-              // Reinstate the interrupt thread
-              Thread.currentThread().interrupt();
-              // Fail with the original exception
-              throw e;
-            }
-          }
+          applyDelay(retries, delayMillisTransactionFailure, e);
         } else {
           throw e;
         }
+      } catch (TransactionException e) {
```

Comment:
Having two catch blocks for SqlTransactionException and Transactionexception looks a bit strange. Looking at SqlTransactionRunner#run when the cause is SqlException you should be getting SqlTransactionException every time the cause is SQLException, should not you?

Reply:
I thought about it earlier, so there are two places where connection failure can occur - 
1. https://github.com/cdapio/cdap/blob/fab9cfd6b79cf3f23f74fa272771e5434b44d8c4/cdap-data-fabric/src/main/java/io/cdap/cdap/spi/data/sql/SqlTransactionRunner.java#L62
2. https://github.com/cdapio/cdap/blob/fab9cfd6b79cf3f23f74fa272771e5434b44d8c4/cdap-data-fabric/src/main/java/io/cdap/cdap/spi/data/sql/SqlTransactionRunner.java#L66-L72

At the first place, we don't throw a sql transaction exception so I had to handle it like this.
wdyt??

Reply:
Discussed offline to through sql transaction exception at both places.